src/test/java/org/broadinstitute/hellbender/tools/walkers/sv/SVAnnotateUnitTest.java

```diff
@@ -0,0 +1,682 @@
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.*;
+
+import static java.util.Objects.isNull;
+
+public class SVAnnotateUnitTest extends GATKBaseTest {
+
+    // Pairs of intervals with different relationships to check if first (variant) interval spans second (feature)
+    // interval (boolean) and count how many of the variant breakpoints (start and end of first interval) are within
+    // the feature (second) interval (integer)
+    @DataProvider(name="variantFeatureComparisons")
+    public Object[][] getVariantFeatureComparisonsData() {
+        return new Object[][] {
+                { new SimpleInterval("chr1", 1, 2000), new SimpleInterval("chr2", 100, 200), false, 0 },
+                { new SimpleInterval("chrX", 6000, 8000), new SimpleInterval("chrX", 30, 400), false, 0 },
+                { new SimpleInterval("chr3", 10000, 30000), new SimpleInterval("chr3", 9000, 20000), false, 1 },
+                { new SimpleInterval("chrY", 500, 700), new SimpleInterval("chrY", 600, 800), false, 1 },
+                { new SimpleInterval("chr22", 40, 90), new SimpleInterval("chr22", 25, 700), false, 2 },
+                { new SimpleInterval("chr19", 33, 33), new SimpleInterval("chr19", 22, 44), false, 2 },
+                { new SimpleInterval("chr10", 900, 4000), new SimpleInterval("chr10", 2000, 2200), true, 0 },
+        };
+    }
+
+    // Check variant / feature interval comparison functions for span and number of breakpoints
+    @Test(dataProvider = "variantFeatureComparisons")
+    public void testVariantFeatureComparisons(
+            final SimpleInterval variantInterval,
+            final SimpleInterval featureInterval,
+            final boolean expectedVariantSpansFeature,
+            final int expectedNumBreakpointsInsideFeature)
+    {
+        final boolean actualVariantSpansFeature = SVAnnotate.variantSpansFeature(variantInterval, featureInterval);
+        Assert.assertEquals(expectedVariantSpansFeature, actualVariantSpansFeature);
+
+        final int actualNumBreakpointsInsideFeature =
+                SVAnnotate.countBreakendsInsideFeature(variantInterval, featureInterval);
+        Assert.assertEquals(expectedNumBreakpointsInsideFeature, actualNumBreakpointsInsideFeature);
+    }
+
+    // Toy transcription start sites and initTree() method for testing annotateNearestTranscriptionStartSite()
+    private static final SVInterval[] transcriptionStartSites = {
+            new SVInterval(0, 100, 101),
+            new SVInterval(1, 150, 151),
+            new SVInterval(1, 200, 201),
+            new SVInterval(1, 250, 251),
+            new SVInterval(2, 1, 2)
+    };
+
+    private static SVIntervalTree<String> initTree() {
+        final SVIntervalTree<String> tree = new SVIntervalTree<>();
+        final String[] genes = {"A", "B", "C", "D", "E"};
+        Assert.assertEquals(transcriptionStartSites.length, genes.length);
```

Comment:
Just a suggestion: `Assert` is designed for testing the main code base, so if this fails one would expect that there is an issue in `SVAnnotate.java`. In this case, it would actually be an error in the way the test itself is coded and throwing a `TestException` would be more appropriate.
```suggestion
        if (transcriptionStartSites.length != genes.length) {
            throw new TestException("Transcription start sites list and genes array are not the same length");
        }
```